rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+4. The ui should be made using bevy components.
+
+### New project construction
+
+1. Handled by `bevy_editor_command`.
+2. Retrieve template from asset repo or local template storage.
+3. Build template object mapping template languages to their result.
+    1. For example foo.rs.tera should become foo.rs.
+4. Fill in appropriate template fields from the user's environment.
+5. Add CI workflows.
+6. Add licenses.
+7. Initialize as Git repo.
+
+### Asset management
+
+1. Template management can be done using terra.
+2. Downloading static assets can be managed by reqwest.
+3. Downloading code assets can be done with cargo.
+4. Downloading template assets can be done with libgit2.
+5. Template assets should use `directories` to store system wide.
+
+### Distributable Packaging
+
+1. Bundle `cargo build` with built asset files.
+2. Let developers create per-target custom build scripts.
+3. Should ideally support WASM out of the box.
+
+### Credential/secrets management
+
+1. Should use the operating system's keychain/credential storage for storing
+   secrets.
+2. For now, only for uploading to itch.io in CI builds.
+
+### Scene/Entity manipulator (with possible component editing dispatch).
+
+1. Can use rfc 62 as its data model.
+2. Should provide an editor for adding and removing components to entities.
+3. Should also contain a basic editor for setting the initial values of an
+   entity's components.
+   1. As a bonus, plugins can allow for domain-specific editor for given
+      component types.
+4. The scene viewer should place all objects in a scene at their initial
+   position.
+5. A play button lets users run any given scene with the world's systems.
+
+### Systems list
+
+1. Can use rfc 62 as its data model.
+2. Can add an optional feature for Bevy's system to store metadata about the
+   source location for systems, this will allow us to even find dynamic systems
+   created at run-time.
+
+### Plugin architecture
+
+1. Most of the features listed above should be implemented as plugins.
+2. This plugin-heavy approach allows future additions of i18n, l10n, and a11y.
+3. a11y should use bevy_a11y
+4. Internationalization can be done with either fluent or
+   rust-i18n.
+    1. If fluent, we should bring bevy_fluent or a similar project into the
+       core project.
+
+## Drawbacks
+
+This is a lot of things for a single RFC. It would be possible to make a minimum
+editor with fewer features, and would lead to an editor quicker.
+
+However, the size of this RFC is merited for these reasons:
+
+1. Most of these features are already required for making the editor usable.
+2. Implementing most of these features can go faster when it assumes the other
+   features are build along side it. For example, the metadata used for the
+   entity manipulator can also be used for systems browser.
+3. Making these features as plug-ins helps us make sure the plugin system is
+   appropriately powerful.
+
+## Rationale and alternatives
```

Comment:
I think that each of these questions should get their own subsection with FAQ style headings. Currently, they're a bit too terse to properly understand.